File: LeetCode/Java/Easy/AverageOfLevelsInBinaryTree.java
package Easy;

import java.util.*;

public class AverageOfLevelsInBinaryTree {
	
	public static void main (String[] args){
		
		
	}
	
	public List<Double> averageOfLevels(TreeNode root) {
		ArrayList<Double> result = new ArrayList<Double>();
		if (root == null) return result;
		Queue<TreeNode> que = new LinkedList<TreeNode>();
		que.offer(root);
		while (!que.isEmpty()){
			double count = que.size();
			double sum = 0;
			for (int i = 0; i < count; i++){
				TreeNode node = que.poll();
				if (node.left != null) que.offer(node.left);
				if (node.right != null) que.offer(node.right);
				sum += node.val;
			}
			result.add(sum/count);
		}
		
		return result;
	}
}

File: LeetCode/Python/Easy/AverageOfLevelsInBinaryTree.py
class Solution:
    def averageOfLevels(self, root):
        result = []
        if root is None:
            return result
        q = collections.deque()
        q.append(root)
        while q:
            count, size = len(q), len(q)
            level_sum = 0
            while count > 0:
                node = q.popleft()
                level_sum += node.val
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
                count -= 1
            result.append(level_sum / float(size))
        return result
